### PR TITLE
Update sanitizer blacklisted attr types

### PR DESCRIPTION
**Pull request by alabiaga**
Blacklist invalid input types `password` `file` in mustache template sanitizer to conform with validator spec that bans these types.

https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L4362

https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L4417

Address vulnerability in b127476355. Note that it doesn't address other concerns and checks in spec that also needs to be made as mentioned in ticket.

### CODE REVIEW

File: validator/validator-main.protoascii [hunk 1]
```diff
@@ -4359,6 +4359,8 @@ tags: {
   }
   attrs: {
     name: "type"
+    # When updating this blacklist be sure to update the sanitizer BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES
+    # as these values also need to be stripped when used in templates.
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: please add a link to the file that BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES appears in

Comment (alabiaga) [reply validator/validator-main.protoascii]: done. Thanks

File: src/sanitation.js [hunk 2]
```diff
@@ -160,10 +160,16 @@ const BLACKLISTED_ATTR_VALUES = [
   /*eslint no-script-url: 0*/ '</script',
 ];
 
-/** @const {!Object<string, !Object<string, !RegExp>>} */
+/**
+ * These values should align with any changes to the blacklisted input types
+ * rules in validator-main.protoascii.
+ * https://github.com/ampproject/amphtml/blob/3150dc2974c49e555e98e7df10c8bad19f488251/validator/validator-main.protoascii
```
Comment (honeybadgerdontcare) [inline src/sanitation.js]: `blob/master/validator`

File: validator/validator-main.protoascii [hunk 3]
```diff
@@ -4359,6 +4359,9 @@ tags: {
...
+    # https://github.com/ampproject/amphtml/blob/3150dc2974c49e555e98e7df10c8bad19f488251/src/sanitation.js#L164
```
Comment (honeybadgerdontcare) [inline validator/validator-main.protoascii]: `blob/master/src/sanitation.js`

**Comment by choumx** [comment]
What's up with the 20 `Merge branch 'master'...` commits? :)

File: src/sanitation.js [hunk 4]
```diff
@@ -160,10 +160,16 @@ const BLACKLISTED_ATTR_VALUES = [
...
+ * These values should align with any changes to the blacklisted input types
+ * rules in validator-main.protoascii.
+ * https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii
+ *
+ * @const {!Object<string, !Object<string, !RegExp>>}
+ */
 const BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES = dict({
   'input': {
-    'type': /(?:image|button)/i,
+    'type': /(?:button|file|image|password)/i,
```
Comment (choumx) [inline src/sanitation.js]: We should still allow this for specs other than AMP4EMAIL.

Comment (alabiaga) [reply src/sanitation.js]: Ah, but this is actually blacklisted from the other specs

https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L4362

but this has not been updated.

Comment (alabiaga) [reply src/sanitation.js]: > What's up with the 20 `Merge branch 'master'...` commits? :)

ah yes, this is my fault as I wrote my commits from merges in my master. I'll fix it in next PRs as it seems to be an issue with all my branches.

Comment (alabiaga) [reply src/sanitation.js]: I just took a look again, I do need to special handle this for just the AMP spec, as there is a rule allowing for as long as the input type file or password is a descendant of a form of method type POST. But for the other specs (AMP4ADs, ACTIONS), it is actually not allowed.

**Comment by alabiaga** [comment]
Just an update. This looks to be more involved than I thought. The sanitizer needs to be retrofitted to be aware of the AMP HTML format it is sanitizing against.  There are rules we define in the validator spec that we don't enforce in the context of mustache templates.

**Comment by choumx** [comment]
That's correct.

**Comment by alabiaga** [comment]
Closing, I will have a design doc for this. Thanks all.

**Pull request closed without merging**